cpp/include/raft/random/rng_launch.cuh

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) 2018-2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef __RNG_LAUNCH_H
+#define __RNG_LAUNCH_H
+
+#pragma once
+
+#include "detail/rng_launch.cuh"
+#include "rng_state.hpp"
+
+namespace raft {
+namespace random {
```

Comment:
I like the separation between `rng_state` and the actual functions which use it to generate distributions. I think we should continue to use the `rng.cuh` as the main entrypoint for the functions, though. Can we just add the contents of this file and add a deprecation message for the individual pieces which we intend to remove?

Reply:
I tried to do this, but it does mean it's a little more tricky to properly ensure that we can separate the device-only API from the rest: you'll see I'm using some macro mechanism to make sure that certain parts get deprecated if included through `rng.cuh` rather than `rng_device.cuh` (which is what you should be using in the future for the device-only API).

Please do check this mechanism again because I might have missed something!